src/main/java/org/prebid/server/handler/NotificationEventHandler.java

```diff
@@ -0,0 +1,143 @@
+        this.analyticsReporter = analyticsReporter;
+        this.trackingPixels = trackingPixels;
+    }
+
+    public static NotificationEventHandler create(AnalyticsReporter analyticsReporter) {
+        final Map<String, TrackingPixel> trackingPixels = new HashMap<>();
+        trackingPixels.put(JPG_FORMAT, TrackingPixel.of(JPG_CONTENT_TYPE, readTrackingPixel(TRACKING_PIXEL_JPG)));
+        trackingPixels.put(PNG_FORMAT, TrackingPixel.of(PNG_CONTENT_TYPE, readTrackingPixel(TRACKING_PIXEL_PNG)));
+        return new NotificationEventHandler(Objects.requireNonNull(analyticsReporter), trackingPixels);
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        final MultiMap queryParameters = context.request().params();
+
+        final NotificationEvent notificationEvent;
+        try {
+            notificationEvent = makeNotificationEvent(queryParameters);
+        } catch (InvalidRequestException ex) {
+            respondWithBadStatus(context, ex.getMessage());
+            return;
+        }
+
+        analyticsReporter.processEvent(notificationEvent);
+
+        final Map<String, String> queryParams = HttpContext.from(context).getQueryParams();
+        final String format = queryParams.get(FORMAT_PARAMETER);
+
+        try {
+            validateEventRequestQueryParams(format);
+        } catch (InvalidRequestException ex) {
+            respondWithBadStatus(context, ex.getMessage());
+            return;
+        }
+
+        respondWithOkStatus(context, format);
+    }
+
+    private static byte[] readTrackingPixel(String path) {
+        try {
+            return ResourceUtil.readByteArrayFromClassPath(path);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(String.format("Failed to load pixel image at %s", path), e);
+        }
+    }
+
+    private NotificationEvent makeNotificationEvent(MultiMap queryParameters) {
+        final String type = queryParameters.get(TYPE_PARAMETER);
+        if (ObjectUtils.notEqual(type, VIEW_TYPE) && ObjectUtils.notEqual(type, WIN_TYPE)) {
+            throw new InvalidRequestException(String.format(
+                    "Type is required query parameter. Possible values are win and view, but was %s", type));
+        }
+
+        final String bidId = queryParameters.get(BID_ID_PARAMETER);
+        if (StringUtils.isBlank(bidId)) {
+            throw new InvalidRequestException("bidid is required query parameter and can't be empty.");
+        }
+
+        final String bidder = queryParameters.get(BIDDER_PARAMETER);
+        if (StringUtils.isBlank(bidder)) {
+            throw new InvalidRequestException("bidder is required query parameter and can't be empty.");
+        }
+        return NotificationEvent.of(type, bidId, bidder);
+    }
+
+    private void validateEventRequestQueryParams(String format) {
+        if (format != null && ObjectUtils.notEqual(format, JPG_FORMAT) && ObjectUtils.notEqual(format, PNG_FORMAT)) {
```

Comment:
This is better to be replaced with `if( format != null && trackingPixels.containsKey(format))`